### ISSUE TITLE
Can we remove the Azure vendor-lock?

### ISSUE THREAD

**Issue by enlightenedpie** [microsoft/code-push-server#7]
"The CodePush Server requires Azure Blob Storage to operate."

"CodePush Server is designed to run as Azure App Service"

I understand that this was developed at Microsoft, therefore a strong tie to Azure is par for the course. However, many dev teams (like mine) would love to use the new standalone CodePush but we are in other cloud infras; AWS in my case. It would be a monumental task to convince our tech leadership and devops team to spin up an account with Azure just to run this. 

Is there a way we can remove the vendor lock by providing our own binary storage and service runners? I'm just asking the question for now, but if it's currently not possible and you're open to PRs I'd be happy to dig through the code and provide a patch to provide external providers.

Thx!

**Comment by lucen-ms** [comment]
Please feel free to fork the repository and add whatever functionality you need. We do not accept feature contributions though.

**Comment by nmengual** [comment]
@enlightenedpie I'm testing to use Redis and S3 instead of Azure services, maybe it can be useful as a base.  https://github.com/nmengual/code-push-server

**Comment by Dylan-yoon-mp** [comment]
> [@enlightenedpie](https://github.com/enlightenedpie) I'm testing to use Redis and S3 instead of Azure services, maybe it can be useful as a base. https://github.com/nmengual/code-push-server

@nmengual When do you think the testing will be completed?

**Comment by htdcx** [comment]
You can also try other components: [code-push-server-go](https://github.com/htdcx/code-push-server-go)

**Comment by Erwan-DATIN** [comment]
Hello everyone, 

I guess I'm not alone, any update would be welcomed.
appcenter shutting down deadline coming so fast I'm not sure I expect that much.

"Expo update" could be an alternative to codepush on the paper.
But it looks like "expo SDK" and "Expo Go app" are too much linked : 
- Expo go update in Apple Store means you have to update your react-native project with matching expo SDK....

**Comment by ssut** [comment]
Hi everyone, I've been working on a project called [code-push-cloudflare-workers](https://github.com/ssut/code-push-cloudflare-workers), a serverless alternative to CodePush built on Cloudflare Workers.
If you're interested, I'm planning to complete this and verify that it works perfectly very near soon. Please take a look!

**Comment by sammiepls** [comment]
> Hi everyone, I've been working on a project called [code-push-cloudflare-workers](https://github.com/ssut/code-push-cloudflare-workers), a serverless alternative to CodePush built on Cloudflare Workers. If you're interested, I'm planning to complete this and verify that it works perfectly very near soon. Please take a look!

Hey, I'm interested in this project! Am I able to try it out locally?

**Comment by ssut** [comment]
> > Hi everyone, I've been working on a project called [code-push-cloudflare-workers](https://github.com/ssut/code-push-cloudflare-workers?rgh-link-date=2025-01-23T06%3A45%3A30.000Z), a serverless alternative to CodePush built on Cloudflare Workers. If you're interested, I'm planning to complete this and verify that it works perfectly very near soon. Please take a look!
> 
> Hey, I'm interested in this project! Am I able to try it out locally?

Yes, it's almost ready to go, and if you want you can run the server locally by executing `wrangler dev` command. For further questions, please open a new issue on the project!
